src/components/BaseLayerSwitcher/BaseLayerSwitcher.js

```diff
@@ -0,0 +1,198 @@
+  return layerImages[nextIndex];
+};
+
+function BaseLayerSwitcher({
+  layers,
+  layerImages,
+  altText,
+  className,
+  titles,
+}) {
+  const baseLayers = layers.filter(layer => layer.getIsBaseLayer());
+  const [switcherOpen, setSwitcherOpen] = useState(false);
+  const [isClosed, setIsClosed] = useState(true);
+  const [currentLayer, setCurrentLayer] = useState(getVisibleLayer(baseLayers));
+
+  /* Images are loaded from props if provided, fallback from layer */
+  const images = layerImages
+    ? Object.keys(layerImages).map(layerImage => layerImages[layerImage])
+    : baseLayers.map(layer => layer.previewImage);
+
+  const openClass = switcherOpen ? ' rs-open' : '';
+  const closedClass = isClosed ? ' rs-closed' : '';
+
+  const onButtonSelect = (open, layer) => {
+    if (!open) {
+      setSwitcherOpen(true);
+      return;
+    }
+    setCurrentLayer(layer);
+    layer.setVisible(true);
+    setSwitcherOpen(false);
+  };
+
+  const setTabIndex = (closed, buttonIndex) => {
+    if (closed) {
+      return buttonIndex === 0 ? '0' : '-1';
+    }
+    return '0';
+  };
+
+  if (!baseLayers || baseLayers.length < 2) {
+    return null;
+  }
+
+  useEffect(() => {
+    /* Ensure correct layer is active on app load */
+    if (currentLayer !== getVisibleLayer(baseLayers)) {
+      setCurrentLayer(getVisibleLayer(baseLayers));
+    }
+  }, [currentLayer]);
+
+  useEffect(() => {
+    /* Used for correct layer image render with animation */
+    let timeout;
+    if (!switcherOpen) {
+      timeout = setTimeout(() => {
+        setIsClosed(true);
+      }, 200);
+    } else {
+      setIsClosed(false);
+    }
+    return () => clearTimeout(timeout);
+  }, [switcherOpen]);
+
+  const toggleBtn =
+    baseLayers.length > 0 ? (
+      <div
+        className="rs-base-layer-switcher-button rs-close"
+        role="button"
+        onClick={() => setSwitcherOpen(false)}
+        onKeyPress={e => e.which === 13 && setSwitcherOpen(false)}
+        tabIndex="0"
+        aria-label={titles.closeSwitcher}
+        title={titles.closeSwitcher}
+      >
+        <FaChevronCircleLeft size={15} focusable={false} />
+      </div>
+    ) : null;
+
+  return (
+    <div className={`${className}${openClass}`}>
+      <div className="rs-base-layer-switcher-layers">
+        {!isClosed && toggleBtn}
+        {baseLayers.map((layer, index) => {
+          const layerName = layer.getName();
+          const buttonTitle = isClosed ? titles.openSwitcher : layerName;
+          const activeClass =
+            layerName === currentLayer.getName() ? ' rs-active' : '';
+          return (
+            <div
+              key={layer.key}
+              className={`rs-base-layer-switcher-button rs-layer${openClass}${activeClass}`}
+              role="button"
+              title={buttonTitle}
+              aria-label={buttonTitle}
+              onClick={() => onButtonSelect(switcherOpen, layer)}
```

Comment:
no need to pass switcherOpen in the parameter of the function.

Reply:
true, I can call it directly